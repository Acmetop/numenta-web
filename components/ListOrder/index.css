/**
 * Numenta shared web components
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

.ol {
  composes: list from 'tachyons';
  composes: mb4 from 'tachyons';
  composes: pl0 from 'tachyons';
}
