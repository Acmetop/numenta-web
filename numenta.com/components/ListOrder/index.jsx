import React from 'react'

import styles from './index.css'


const ListOrder = ({children, marker}) => {
  const childrenWithProps = React.Children.map(children, (child) =>
    React.cloneElement(child, {marker})
  )

  return (
    <ol className={styles.ol}>
      {childrenWithProps}
    </ol>
  )
}

ListOrder.propTypes = {
  children: React.PropTypes.any.isRequired,
  marker: React.PropTypes.oneOf(['decimal', 'disc']),
}

export default ListOrder
