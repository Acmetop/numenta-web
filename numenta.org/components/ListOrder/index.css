/**
 * Numenta.org HTM Community website source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

.ol {
  composes: list from 'tachyons';
  composes: mb4 from 'tachyons';
  composes: pl0 from 'tachyons';
}
