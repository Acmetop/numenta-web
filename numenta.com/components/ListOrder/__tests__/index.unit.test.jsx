import React from 'react'
import renderer from 'react-test-renderer'

import ListOrder from '../../ListOrder'


describe('<ListOrder /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <ListOrder />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
