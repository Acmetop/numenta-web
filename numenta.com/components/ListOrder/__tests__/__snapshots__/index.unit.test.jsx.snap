exports[`<ListOrder /> React component unit test suite Renders correctly 1`] = `
<ol
  className="ol" />
`;
